mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -383,6 +383,21 @@ abstract class PcCollector[T](driver: InteractiveDriver, params: OffsetParams):
     trees.collect { case t: Tree =>
       t
     }
+
+  // NOTE: Connected to https://github.com/lampepfl/dotty/issues/16771
+  // `sel.nameSpan` is calculated incorrectly in (1 + 2).toString
+  // See test DocumentHighlightSuite.select-parentheses
+  private def selectNameSpan(sel: Select)(using Context): Span =
+    val span = sel.span
+    val realName = sel.name.stripModuleClassSuffix.lastPart
+    if span.exists then
+      val point = span.point
+      if sel.name.toTermName == nme.ERROR then Span(point)
+      else if sel.qualifier.span.start - span.start >= realName.length
+      then // right associative
+        Span(span.start, span.start + realName.length, point)
+      else Span(point, span.end, point)
+    else span
```

Comment:
```suggestion
  private def selectNameSpan(sel: Select)(using Context): Span =
    val span = sel.span
    if span.exists then
      val point = span.point
      if sel.name.toTermName == nme.ERROR then Span(point)
      else if sel.qualifier.span.start > span.point
      then // right associative
        val realName = sel.name.stripModuleClassSuffix.lastPart
        Span(span.start, span.start + realName.length, point)
      else Span(point, span.end, point)
    else span
```
I think we should have `sel.qualifier.span.start > span.point` instead and that should work reliably. point points to the place between qual and apply.

Otherwise, this will not work for larger amounts of parens, no?

We can most likely just raise the PR with a fix to Dotty, could you take a look into that?

Reply:
This way looks more reliable, thanks!